src/Application/Service/Property/PropertyViewRequest.js

```diff
@@ -1,21 +1,47 @@
-function PropertyViewRequest(propertyId,
-  ownerId) {
+import { IsBlankStringSpecification } from '@Application/Specification/IsBlankStringSpecification';
+import { IsNumberSpecification } from '@Application/Specification/IsNumberSpecification';
+
+class PropertyViewRequestBuilder {
+  constructor() {
+    this.propertyId = null;
+    this.ownerId = null;
+  }
+
+  withPropertyId(propertyId) {
     this.propertyId = propertyId;
+    return this;
+  }
+
+  withOwnerId(ownerId) {
     this.ownerId = ownerId;
+    return this;
+  }
 
-    if (this.constructor === PropertyViewRequest) {
-      Object.freeze(this);
+  _assertThatPropertyIdIsValid() {
+    const isNumber = new IsNumberSpecification();
+
+    if (!isNumber.isSatisfiedBy(this.propertyId)) {
+      throw new Error('propertyId must be a number');
     }
-    console.log(this.propertyId)
   }
 
-PropertyViewRequest.propertyId = function propertyId() {
-  return this.propertyId;
-};
+  _assertThatOwnerIdIsValid() {
+    const isBlankString = new IsBlankStringSpecification();
 
-PropertyViewRequest.ownerId = function ownerId() {
-  return this.ownerId;
-};
+    if (isBlankString.isSatisfiedBy(this.ownerId)) {
+      throw new Error('ownerId must be a non-blank string');
+    }
+  }
 
-module.exports = PropertyViewRequest
+  build() {
```

Comment:
Im not used to use `build`. Why not check if object is valid in the constructor instead of having to create an additional method to check it?

Reply:
to move the responsibility of type checking outside the domain model and make it a bit clearer

Reply:
Im not fan of having an additional method to check types, but I can not find a better solution (only using typescript)